Safety/Src/PPM.cpp

```diff
@@ -131,22 +135,57 @@ void HAL_TIM_IC_CaptureCallback(TIM_HandleTypeDef *htim)
 
 	if (htim->Instance == TIM15)
 	{
-
-		static volatile uint8_t index = 0;
-
-		volatile uint32_t time_diff = HAL_TIM_ReadCapturedValue(htim, TIM_CHANNEL_1);
-		__HAL_TIM_SET_COUNTER(htim, 0);
-
-		float pulseLength = counter_to_time(time_diff, htim->Init.Prescaler) - PULSE_WIDTH;
-
-		if (pulseLength > MIN_WIDTH_OF_RESET_PULSE)
+		if(htim->Channel == HAL_TIM_ACTIVE_CHANNEL_1)
 		{
-			index = 0;
+			static volatile uint8_t index = 0;
+
+			volatile uint32_t time_diff = HAL_TIM_ReadCapturedValue(htim, TIM_CHANNEL_1);
+			__HAL_TIM_SET_COUNTER(htim, 0);
+
+			//Make sure we don't lose track of the PWM input that uses the same timer
+			pwmTicks += time_diff - pwmStartTime;
+			if (time_diff < pwmStartTime)
+			{
+				pwmTicks += 0xffff; //Handle timer overflow
+			}
+			pwmStartTime = 0;
+
+			float pulseLength = counter_to_time(time_diff, htim->Init.Prescaler) - PULSE_WIDTH;
+
+			if (pulseLength > MIN_WIDTH_OF_RESET_PULSE)
+			{
+				index = 0;
+			}
+			else if (index < MAX_PPM_CHANNELS)
+			{
+				ppm_values[index] = pulseLength;
+				index++;
+			}
 		}
-		else if (index < MAX_PPM_CHANNELS)
+		else if (htim->Channel == HAL_TIM_ACTIVE_CHANNEL_2) //Recieve RSSI PWM
```

Comment:
This will never be called unless an appropriate `HAL_TIM_IC_Start_IT(&htim15, TIM_CHANNEL_2);` is also called during the init process.